projects/fuzzing-ethereum-network.md

```diff
@@ -0,0 +1,107 @@
+
+## Motivation
+
+In order to ensure the good running of Ethereum, it is imperative that is remains highly secure. As a large blockchain, Ethereum represents a target for potential hackers. 
+
+The Ethereum network and the differents protocols of the devp2p, plays an essential role in enabling communication between nodes, an essential element for the operation of the protocol. 
+
+Vulnerabilities or errors in this network could cause significant problems. It is therefore crucial to guarantee the maximum security of the network and exchanges between nodes.
+
+## Project description
+
+To increase security, I propose the creation of fuzzers, with the aim of potentially spotting vulnerabilities and bugs present on the Ethereum network and clients implementing the devp2p.
+
+The project was originally proposed by [Fredrik](https://github.com/fredriksvantes).
+
+The fuzzers will allow arbitrary/specific data to be sent to the various ethereum nodes through its network to cause unexpected behavior such as crashes, memory leaks and others...
+
+These behaviors will then be reported in order to be able to correct them and secure the network.
+
+## Specification
+
+
+The realization of my solution requires sending of messages using the fuzzers that I will develop. The process is broken down into two fundamental steps:
+- Connect to the ethereum network and send messages
+- Develop different fuzzers and send malicious inputs to the network.
+
+
+To build the solution I choosed to use the Rust programming language :
+Rust is a good solution for the development of fuzzer, he have great performance. Performance is a key for fuzzer, they need to generate and send a great quantity of inputs . Rust have other advantages like a good ecosystem, security of memory...
+
+Then in order to connect to the different protocols that form the devp2p I had the choice between two options:
+
+- implement the devp2p specifications myself in order to connect to the ethereum network
+- use and edit an existing implementation of the devp2p.
+
+For time reasons I decided to choose the second option:
+
+I chose to use the reth project, specifically the network module of reth .
+Why reth? for several reasons :
+- reth is an implementation of ethereum in rust , the same language I chose to use for the development of fuzzers this will allow me
+to use a single language.
+- reth is fast and very efficient, it will allow me to be able to spam with my fuzzers.
+- the documentation of reth, especially on the network part (network crate) is well detailed.
+
+So once I have correctly integrated the devp2p implementation by modifying the network component of reth , I will be able to
+communicate through the ethereum network and be able to start integrating my fuzzers.
+
+For the fuzzing, there are a lot of fuzzing techniques, the goal will be to identify the most appropriate to integrate them in priority and then to use the maximum of different techniques in order to have the most results.
```

Comment:
you might want to look into network upgrades and changes in eth protocol, creating payloads to reflect specific changes